src/java/org/apache/cassandra/db/guardrails/Values.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.guardrails;
+
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+import com.google.common.collect.Sets;
+
+import org.apache.cassandra.service.ClientState;
+
+import static java.lang.String.format;
+
+/**
+ * A guardrail that warns about but ignores some specific values, and rejects the use of some other values.
+ *
+ * @param <T> The type of the values of which certain are disallowed.
+ */
+public class Values<T> extends Guardrail
+{
+    private final Function<ClientState, Config<T>> configProvider;
+    private final String what;
+
+    /**
+     * Creates a new values guardrail.
+     *
+     * @param configProvider a {@link ClientState}-based provider of {@link Config}s.
+     * @param what           The feature that is guarded by this guardrail (for reporting in error messages).
+     */
+    public Values(Function<ClientState, Config<T>> configProvider, String what)
+    {
+        this.configProvider = configProvider;
+        this.what = what;
+    }
+
+    /**
+     * Checks whether the provided value would trigger this guardrail.
+     *
+     * <p>This method is optional (does not have to be called) but can be used in the case some of the arguments
+     * to the actual guardrail method are expensive to build to save doing so in the common case of the guardrail not
+     * being triggered.
+     *
+     * @param value The value to test.
+     * @param state The client state, used to skip the check if the query is internal or is done by a superuser.
+     * @return {@code true} if {@code value} is not allowed by this guardrail,
+     * {@code false otherwise}.
+     */
+    public boolean triggersOn(T value, @Nullable ClientState state)
+    {
+        if (!enabled(state))
+            return false;
+
+        Config<T> config = configProvider.apply(state);
+        return config.getIgnored().contains(value) || config.getDisallowed().contains(value);
+    }
```

Comment:
deadcode; can we remove?

Reply:
Removed, we'll probably add them back to the framework when have the use cases for them.